File: 2020/day/6/Main.hs
import Text.ParserCombinators.ReadP
import Data.Char
import Data.List

parse = munch1 isAlpha `endBy` char '\n' `sepBy` char '\n'

main = do
    input <- readFile "input.txt"
    let [(groups, "")] = readP_to_S (parse <* eof) input

    -- Part 1
    print $ length $ concatMap (foldr union []) groups

    -- Part 2
    print $ length $ concatMap (foldr1 intersect) groups

File: 2020/day/6/main.py
import fileinput

group = []
groups = []

for line in fileinput.input('input.txt'):
    line = line.strip()

    if not line:
        groups.append(group)
        group = []
    else:
        group.append(line)

groups.append(group)

ans1 = 0
ans2 = 0

for i, group in enumerate(groups):
    y = set(a for p in group for a in p)
    ans1 += len(y)

    for p in group:
        y &= set(p)

    ans2 += len(y)


print(ans1)
print(ans2)